app/src/main/java/dmar/akan/akademiaandroida/features/characters/data/local/model/CharacterCached.kt

```diff
@@ -0,0 +1,49 @@
+package dmar.akan.akademiaandroida.features.characters.data.local.model
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import dmar.akan.akademiaandroida.features.characters.domain.model.Character
+
+@Entity
+data class CharacterCached(
+    @PrimaryKey
+    val id: Int,
+    val name: String,
+    val species: String,
+    val type: String,
+    val gender: String,
+    val origin: Character.CharacterOrigin,
+    val location: Character.CharacterLocation,
+    val image: String,
+    val episode: List<String>,
+    val url: String
+) {
+
+    constructor(character: Character) : this(
+        character.id,
+        character.name,
+        character.species,
+        character.type,
+        character.gender,
+        character.origin,
```

Comment:
Z racji tego, że zmieniliśmy pola `origin` i `location` na typy pochodzące z warstwy danych, musimy teraz je utworzyć korzystając z konstruktora z wcześniej zdefiniowanego konstruktora.

```suggestion
        CharacterOriginCached(character.origin),
```